do_command.c

```diff
@@ -405,14 +405,43 @@ child_process(entry *e, user *u) {
 			 * up the mail command and subjects and stuff...
 			 */
 			if (mailto != NULL) {
-				char	mailcmd[MAX_COMMAND];
-
-				if (strlens(MAILFMT, MAILARG, NULL) + 1
-				    >= sizeof mailcmd) {
-					fprintf(stderr, "mailcmd too long\n");
+				char mailcmd[MAX_COMMAND] = "";
+				const char *msg = NULL;
+
+				if (Mailer != NULL) {
+					if (strstrcount(Mailer, "%s") == 1) {
+						if (strlens(Mailer, mailto) + 1
```

Comment:
I think you need a NULL terminator for strlens() here.

Reply:
check it now please? i hope to have resolved your concerns.